src/cargo/ops/vendor.rs

```diff
@@ -297,7 +297,7 @@ fn sync(
                     // Can be removed once the minimal supported Windows version gets bumped.
                     tracing::warn!("failed to `mv {unpacked_src:?} {dst:?}`: {e}");
                     let paths: Vec<_> = walkdir(&unpacked_src).map(|e| e.into_path()).collect();
-                    cp_sources(pkg, src, &paths, &dst, &mut file_cksums, &mut tmp_buf, gctx)
+                    cp_sources(pkg, unpacked_src.as_path(), &paths, &dst, &mut file_cksums, &mut tmp_buf, gctx)
```

Comment:
nit: personally perfer letting compiler deref

```suggestion
                    cp_sources(pkg, &unpacked_src, &paths, &dst, &mut file_cksums, &mut tmp_buf, gctx)
```

And we'll need to run `cargo fmt`